internal/indexer/indexer.go

```diff
@@ -88,6 +91,13 @@ func (i *Indexer) ProcessTransaction(ctx context.Context, transaction ingest.Led
 			return fmt.Errorf("processing effects state changes: %w", err)
 		}
 		i.Buffer.PushStateChanges(effectsStateChanges)
+
+		// 2.2. Index contract deploy state changes
+		contractDeployStateChanges, err = i.contractDeployProcessor.ProcessOperation(ctx, opParticipants.OpWrapper)
```

Comment:
Small design idea: since we are using the same interface here, we can instead have a list with these two processors and use a for loop to push the state changes. So something like this:

```
operationProcessors []OperationProcessorInterface

for _, processor: range b.operationProcessors:
     stateChanges, err := processor.ProcesssOperation...
     ....
     i.Buffer.PushStateChanges(stateChanges)
```

Reply:
I agree with that change but I think it's outside the scope of this PR. It should be done in a dedicated PR.

Reply:
I opened a PR [here](https://github.com/stellar/wallet-backend/pull/254), but in reality this won't scale for more custom contract processors.

[This discussion](https://stellarfoundation.slack.com/archives/C030Z9EHVQE/p1753204684005249) points to the necessity of calling RPC to assert contract interfaces and likely adding a special cache for contracts. This cache can expand a lot and we'll probably need some space-efficient filter, like BloomFilter.

Reply:
I'm leaving that PR in draft. Feel free to either close it or build on top of it.